lib/ansible/executor/task_executor.py

```diff
@@ -875,7 +876,7 @@ def _poll_async_result(self, result, templar, task_vars=None):
 
         async_task = Task.load(dict(
             action='async_status',
-            args={'jid': async_jid},
+            args={'jid': results_file or async_jid},
```

Comment:
I still think this should be 2nd key and not overload jid, otherwise code LGTM

Reply:
Hrm, I'm going to put this back in draft for a bit.  I need to contemplate the windows side of things as well.